packages/core/src/browser/label-provider.ts

```diff
@@ -154,6 +196,122 @@ export class DefaultUriLabelProviderContribution implements LabelProviderContrib
     protected getUri(element: URI | URIIconReference): URI | undefined {
         return URIIconReference.is(element) ? element.uri : element;
     }
+
+    registerFormatter(formatter: ResourceLabelFormatter): Disposable {
+        this.formatters.push(formatter);
+        this.onDidChangeEmitter.fire({
+            affects: (element: URI) => this.canHandle(element) === 1
+        });
+        return Disposable.create(() => {
+            this.formatters = this.formatters.filter(f => f !== formatter);
+            this.onDidChangeEmitter.fire({
+                affects: (element: URI) => this.canHandle(element) === 1
+            });
+        });
+    }
+
+    getUriLabel?(uri: URI): string {
```

Comment:
it should be part of  `getName` and `getLongName`, that formatters are applied properly to all existing client without introducing new APIs

Reply:
We should care about these methods in `WorkspaceUriLabelProviderContribution` as well.

Reply:
done